scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/view/FilterView.java

```diff
@@ -162,26 +188,224 @@ public void clearCriteria() {
         addRow();
     }
 
+    public String generateFilterJsonString() {
+        JSONObject jsonObject = new JSONObject();
+        int widgetCount = filterPanel.getMembersLength();
+        JSONArray constraintsArray = new JSONArray();
+        for (int i = 0; i < widgetCount; i++) {
+            RowFilter row = (RowFilter) filterPanel.getMember(i);
+            String field = row.fieldsList.getSelectedValue();
+            String action = row.actionList.getSelectedValue();
+            String value = row.getValue();
+
+            JSONObject constraintObject = new JSONObject();
+            constraintObject.put("field", new JSONString(field.toString()));
```

Comment:
'toString()' call is redundant here, as "field", "action" and "value" are already of type string

Reply:
True